enhancements/update/configurable-clusterversion-signature-stores.md

```diff
@@ -0,0 +1,164 @@
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+CVO currently fetches the upstream urls for signature verification through [cluster-update-keys]. 
+To increase flexibility and improve disconnected cluster user experience, we need to provide a way for the CVO to 
+get the required signatures from custom signature stores or the OSUS running in the disconnected environment.
+
+
+## Motivation
+
+Today, in the disconnected environments, a user has to clone the signatures config map as well as the release images 
+then apply the configMap to each cluster for the update to proceed successfully.
+With [serve image signatures for disconnected environments], OSUS will be able to serve the cloned image signatures locally.
+
+Currently, there is no way for the CVO to fetch the signatures through custom URIs. This enhancement defines an API change in the CVO
+to accept a custom update-service URI which can be used to verify release signatures. 
+
+
+### User Stories
+
+* As a cluster administrator, I want the cluster's CVO to add new signature store URIs.
+* As a cluster administrator, I want the cluster's CVO to replace the existing signature store URIs with a user provided URI
+
+### Goals
+
+* To verify the release image signature in disconnected environments using OSUS signatures URI.
+* To replace existing signature stores with custom user-provided stores.
+
+### Non-Goals
+
+* This enhancement does not cover how OSUS running in the disconnected environment gets the signatures and if the 
+signatures are validated during the mirroring process.
+
+## Proposal
+
+### Workflow Description
+
+1. Cluster admin will have to add the verification URI to CVO
+2. CVO will reach the newly added URI to verify the release images.
+
+### API Extensions
+
+* CVO will grow a new field, an array, which will be populated with the image verification URI.
+* The api extension will replace the existing signature stores. 
+* To use existing stores, admin will have to add them manually with locally hosted stores, although locally hosted stores 
+are not required for this functionality to work.
+* This will be an array as it'll allow the admin to use locally configured stores with the existing stores
+* Admins will still be able to use the existing config maps for signature verification. 
+
+This enhancement proposes to modify the [api/config/v1/types_cluster_version.go](https://github.com/openshift/api/blob/master/config/v1/types_cluster_version.go) as following,
+
+A new field `SignatureStores` will be added to the [`ClusterVersionSpec`](https://github.com/openshift/api/blob/master/config/v1/types_cluster_version.go#L40-L96) structure:
+
+```go
+type ClusterVersionSpec struct {
+	...
+	// SignatureStores contains the upstream URIs to verify release signatures.
+	//
+	// Valid values are: "{}" or array of upstream signature store string.
+	// Defaults to "{}".
```

Comment:
What is `{}`?  I'd expect an `// +optional` kubebuilder comment for this new property, but expect all the entries in the array to be HTTP(S?) URIs.  Looks like we already have [a `URL` type][1] we could use here, so:

```go
// +optional
SignatureStores []URL `json:"signatureStores,omitempty"`
```

or some such?  And if folks want to add some regular expression validation to that `URL` type, that could happen orthogonally?

[1]: https://github.com/openshift/api/blob/bbe1a26bbafcc8f3abcc72418d8d9492a68d44c6/config/v1/types_cluster_version.go#L531-L532

Reply:
also worth mentioning that if this property is empty, default stores will be used, similar to [how we handle `upstream` today].

[1]: https://github.com/openshift/api/blob/bbe1a26bbafcc8f3abcc72418d8d9492a68d44c6/config/v1/types_cluster_version.go#L73-L74

Reply:
added